### PR TITLE
feat: Add metadata to workout samples

### PR DESCRIPTION
**Pull request by thalysviana**
## Description

This PR adds the following changes

- Increases the minimum deployment target to v14.0
- Adds support to workout metadata
- Replaces workout creation through `HKWorkout` class with `HKWorkoutBuilder` (`HKWorkout` initialize will be deprecated in iOS v17)

Fixes # (issue)

## Type of change

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -385,4 +382,67 @@ export enum WorkoutActivityType {
...
+  doubleValue: number;
+}
+
+export enum WaterSalinityType {
+  freshWater = 0,
+  saltWater = 1,
+}
+
+export type WorkoutMetadata = {
+    // [WorkoutMetadataKey.ActivityType]?: string; // ?
```
Comment (thalysviana) [inline index.ts:421]: I left some of the keys commented due to missing documentation from Apple. Most of them are related to iOS 17, and the documentation is on beta.

File: v2/RNHealthKitCore/HealthKitCore/Workout/WorkoutQueries.swift [hunk 2]
```diff
@@ -35,25 +35,51 @@ public extension HealthKitCore {
...
+    ///   - metadata: Optional metadata for the workout.
     /// - Throws: An error if the workout data cannot be saved to HealthKit.
     func saveCompletedWorkout(
         activityType: HKWorkoutActivityType,
         startDate: Date,
         endDate: Date,
         totalEnergyBurned: Double?,
-        totalDistance: Double?
-    ) async throws {
-        let workout = HKWorkout(
```
Comment (thalysviana) [inline v2/RNHealthKitCore/HealthKitCore/Workout/WorkoutQueries.swift:46]: Apple is deprecating `HKWorkout` initializer on iOS 17. I replaced with `HKWorkoutBuilder` which has support from iOS 12+.

Comment (GGGava) [reply v2/RNHealthKitCore/HealthKitCore/Workout/WorkoutQueries.swift:46]: cool

File: v2/RNHealthKitCoreApp/RNHealthKitCoreApp/ContentView.swift [hunk 3]
```diff
@@ -11,68 +11,43 @@ struct ContentView: View {
...
+    func saveWorkoutWithMetadata(core: HealthKitCore) async {
+        let unit = HKUnit(from: "min")
+        let rawMetadata: [String: Any] = [
+            "HKIndoorWorkout": true,
+            "HKFitnessMachineDuration": ["unit": "min", "doubleValue": 30.0],
+        ]
+
+        let processedMetadata = try! WorkoutHelper.processWorkoutMetadata(rawMetadata)
+
+        dump(try! await core.saveCompletedWorkout(
```
Comment (GGGava) [inline v2/RNHealthKitCoreApp/RNHealthKitCoreApp/ContentView.swift:42]: what is this dump function?

File: index.ts (hunk 1 above)
Comment (GGGava) [reply index.ts:421]: That's fine. Are these types not available yet?

**Review by GGGava — APPROVED** [review]

File: v2/RNHealthKitCoreApp/RNHealthKitCoreApp/ContentView.swift (hunk 3 above)
Comment (thalysviana) [reply v2/RNHealthKitCoreApp/RNHealthKitCoreApp/ContentView.swift:42]: It's for debugging purposes only. But usually I use to inspect objects/structs in a nicer format. It also shows me the hierarchy and private properties.

**Pull request merged**